sensor/RouterChangeSensor.js

```diff
@@ -0,0 +1,64 @@
+/*    Copyright 2020 Firewalla INC 
+ *
+ *    This program is free software: you can redistribute it and/or  modify
+ *    it under the terms of the GNU Affero General Public License, version 3,
+ *    as published by the Free Software Foundation.
+ *
+ *    This program is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU Affero General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Affero General Public License
+ *    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+'use strict';
+
+const log = require('../net2/logger.js')(__filename);
+const Sensor = require('./Sensor.js').Sensor;
+const sem = require('../sensor/SensorEventManager.js').getInstance();
+const rclient = require('../util/redis_manager.js').getRedisClient();
+const Mode = require('../net2/Mode.js');
+const sysManager = require('../net2/SysManager.js');
+const HostTool = require('../net2/HostTool.js');
+const hostTool = new HostTool();
+const modeManager = require('../net2/ModeManager.js');
+
+const routerChangeKey = 'sys:router:change';
+const routerMacKey = 'sys:router:mac';
+
+class RouterChangeSensor extends Sensor {
+  constructor() {
+    super();
+  }
+
+  async run() {
+    sem.once('IPTABLES_READY', async() => {
+      await this.checkRouteMac();
+    });
+    sem.on('NewDeviceFound', async (event) => {
+      await this.checkRouteMac();
+    });
+  }
+
+  async checkRouteMac() {
+    if (!Mode.isSpoofModeOn()) {
+      return;
+    }
+
+    const gatewayIp = sysManager.myGateway();
+    const gatewayMac = await hostTool.getMacByIP(gatewayIp);
+    const isMember = await rclient.sismemberAsync(routerMacKey, gatewayMac);
+    const hasAny = await rclient.scardAsync(routerMacKey);
+    if (!isMember && hasAny > 0) {
+      await modeManager.setNoneAndPublish();
```

Comment:
turn off monitoring on a specific network might be a better solution
use policy targeting `network:<uuid>`, check https://github.com/firewalla/firewalla/blob/cc204133ef5f9c425e702b3077fd4f6886e381cd/controllers/netbot.js#L247

Reply:
Also, we should send a notification to App as well. With this flag only, user won't get notified until he/she opens the app.